Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -312,12 +312,17 @@ void SoftwareGLContext::gl_end()
 
     // Set up normals transform by taking the upper left 3x3 elements from the model view matrix
     // See section 2.11.3 of the OpenGL 1.5 spec
-    auto const& mv_elements = m_model_view_matrix.elements();
-    auto normal_transform = FloatMatrix3x3(
-        mv_elements[0][0], mv_elements[0][1], mv_elements[0][2],
-        mv_elements[1][0], mv_elements[1][1], mv_elements[1][2],
-        mv_elements[2][0], mv_elements[2][1], mv_elements[2][2]);
-    normal_transform = normal_transform.inverse();
+    auto const& inverse_of_model_view_matrix = m_model_view_matrix.inverse();
```

Comment:
I believe the spec actually says to invert the upper left 3x3 sub matrix. Here we are taking the upper left 3x3 matrix of the inverted 4x4 matrix... so we should first take the sub matrix and then do the inverse.